src/pdf/ShoppingList/ShoppingListPDF.tsx

```diff
@@ -0,0 +1,223 @@
+    },
+    tableItemDescription: {
+        width: "25%",
+    },
+    tableQuantity: {
+        width: "35%",
+    },
+    tableNotes: {
+        width: "30%",
+    },
+    tableDone: {
+        width: "10%",
+    },
+    title: {
+        fontSize: "30pt",
+        fontFamily: "Helvetica-Bold",
+    },
+    subtitle: {
+        fontSize: "24pt",
+        fontFamily: "Helvetica",
+    },
+    keyText: {
+        fontSize: "12pt",
+        fontFamily: "Helvetica-Bold",
+    },
+    normalText: {
+        fontSize: "12pt",
+        fontFamily: "Helvetica",
+    },
+});
+
+const OneLine: React.FC<{ header: string; value: string }> = ({ header, value }) => {
+    return (
+        <Text style={[styles.keyText, { textTransform: "capitalize" }]}>
+            {header}: <Text style={styles.normalText}>{value}</Text>
+        </Text>
+    );
+};
+
+const TableHeadings: React.FC<{}> = () => {
+    return (
+        <View style={[styles.flexRow, styles.tableRow]}>
+            <View style={styles.tableItemDescription}>
+                <Text style={styles.keyText}>Item Description</Text>
+            </View>
+            <View style={styles.tableQuantity}>
+                <Text style={styles.keyText}>Quantity</Text>
+            </View>
+            <View style={styles.tableNotes}>
+                <Text style={styles.keyText}>Notes</Text>
+            </View>
+            <View style={styles.tableDone}>
+                <Text style={styles.keyText}>Done</Text>
+            </View>
+        </View>
+    );
+};
+
+const ItemToRow: React.FC<Item> = (item) => {
+    return (
+        <View style={[styles.flexRow, styles.tableRow]}>
+            <View style={styles.tableItemDescription}>
+                <Text style={styles.normalText}>{item.description}</Text>
+            </View>
+            <View style={styles.tableQuantity}>
+                <Text style={styles.normalText}>{item.quantity}</Text>
+            </View>
+            <View style={styles.tableNotes}>
+                <Text style={styles.normalText}>{item.notes}</Text>
+            </View>
+            <View style={styles.tableDone}></View>
+        </View>
+    );
+};
+
+const DisplayItemsList: React.FC<{ itemsList: Item[] }> = ({ itemsList }) => {
+    return (
+        <View>
+            {itemsList.map((item, index) => (
+                <ItemToRow {...item} key={index} />
+            ))}
+        </View>
+    );
+};
+
+const formatCamelCaseKey = (objectKey: string): string => {
+    return objectKey.replaceAll(/([a-z])([A-Z])/g, "$1 $2");
+};
+
+const DisplayAsBlock: React.FC<BlockProps> = <Prop extends BlockProps>(data: Prop) => {
+    return (
+        <View style={[styles.flexColumn, styles.infoCell]}>
+            {Object.keys(data).map((propKey, index) => (
+                <OneLine key={index} header={formatCamelCaseKey(propKey)} value={data[propKey]} />
+            ))}
+        </View>
+    );
+};
+
+const FormatClientCell: React.FC<{ propKey: string; propValue: string }> = ({
```

Comment:
you can probably just rename this to `key` and `value`

Reply:
gives an error - i think `key` is a reserved word